crates/tbdex/src/http/exchanges.rs

```diff
@@ -0,0 +1,319 @@
+                            MessageKind::Quote => {
+                                if let Ok(quote) = serde_json::from_value::<Quote>(value.clone()) {
+                                    Ok(ReplyToMessage::Quote(Arc::new(quote)))
+                                } else {
+                                    Err(serde::de::Error::custom("failed to deserialize quote"))
+                                }
+                            }
+                            MessageKind::OrderStatus => {
+                                if let Ok(order_status) =
+                                    serde_json::from_value::<OrderStatus>(value.clone())
+                                {
+                                    Ok(ReplyToMessage::OrderStatus(Arc::new(order_status)))
+                                } else {
+                                    Err(serde::de::Error::custom(
+                                        "failed to deserialize order_status",
+                                    ))
+                                }
+                            }
+                            MessageKind::Close => {
+                                if let Ok(close) = serde_json::from_value::<Close>(value.clone()) {
+                                    Ok(ReplyToMessage::Close(Arc::new(close)))
+                                } else {
+                                    Err(serde::de::Error::custom("failed to deserialize close"))
+                                }
+                            }
+                            _ => Err(serde::de::Error::custom(format!(
+                                "unexpected message kind {:?}",
+                                kind
+                            ))),
+                        },
+                        Err(_) => Err(serde::de::Error::custom(format!(
+                            "unexpected message kind {}",
+                            k
+                        ))),
+                    },
+                    None => Err(serde::de::Error::custom(format!(
+                        "unexpected message kind {:?}",
+                        kind_str
+                    ))),
+                }
+            }
+
+            fn visit_none<E>(self) -> std::result::Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Err(serde::de::Error::custom("message is missing"))
+            }
+        }
+
+        deserializer.deserialize_option(MessageVisitor)
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct ReplyToRequestBody {
+    pub message: ReplyToMessage,
+}
+impl JsonDeserializer for ReplyToRequestBody {}
+impl JsonSerializer for ReplyToRequestBody {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+
+    #[derive(Debug, serde::Deserialize)]
+    pub struct TestVector<T> {
+        pub input: String,
+        pub output: T,
+    }
+
+    #[test]
+    fn order_update_exchange_request_body() {
```

Comment:
I was able to re-use the test vectors here for testing our multi-types of `WalletUpdateMessage` and `ReplyToMessage` -- @nitro-neal this may be considered poor practice? It was a convenient source of valid test data so 🤷

Reply:
I think this is fine, if test vector changes it may break in 2 places, but thats fine just update in both